op5build/ci_config.yml

```
@@ -1,3 +1,38 @@
-build:
-  exclude: ['el6/x86_64']
+pre:
+  steps: |
+    set -ex
+    rm -rf venv dist
 
+    python3 -m venv venv
+    source venv/bin/activate
+    pip install --upgrade pip
+    pip install poetry wheel==0.34.2 pytest
+
+    cd src
+
+    # We build all the wheels now while we can access PyPI and github repos
+    poetry build -f wheel  # writes to dist/
+    pip wheel -r requirements.txt -w dist/
+
+    # Install all dependencies to run tests
+    pip install dist/*.whl
+    python -m pytest
+
+    # Also include the current pip version so it can be used in the installation
+    pip wheel pip -w dist/
+
+post:
+  steps: |
+    set -ex
+    # Smoke test
+    /opt/plugins/check_aws.py --help
+
+    if [ \${RHEL_VERSION:-0} -ge 8 ]; then
+      # Run unit tests using the installed check_aws package and dependencies
+      # from rpm repos. Install test tools from dist/, created in pre step above
+      # that was copied here from the build system.
+      mv check_aws{,.src}  # don't let Python find this
```

This could perhaps be skipped by running the tests with Python's `-I` option.

---

I tried some things now. Unfortunately python won't find the things we just installed with pip, since they end up in `/usr/local/...` by default, if invoked with `-I`.
However, since pytest is invoked now with `/usr/local/bin/pytest` script, it won't add `.` to `sys.path` anyway, so the line above actually isn't required. But I'll keep it anyway, since if someone changes how python is invoked in the future it would silently use the wrong `check_aws` package.

---

I think I'll change to invoke pytest with `python3 -m pytest` though, for consistency.

---

Aha, okay. :+1: